File: LeetCode/cpp/506.cpp
class Solution {
    class Athlete {
    public:
        int score;
        int pos;
    };

public:
    vector<string> findRelativeRanks(vector<int> &nums) {
        int n = nums.size(), j = 0;
        vector<Athlete> athletes(n, Athlete());
        vector<string> ret(n, "");
        vector<string> medal{"Gold Medal", "Silver Medal", "Bronze Medal"};
        for (int i = 0; i < n; ++i) {
            athletes[i].score = nums[i];
            athletes[i].pos = i;
        }
        sort(athletes.begin(), athletes.end(), [](const Athlete &a1, const Athlete &a2) {
            return a1.score >= a2.score;
        });
        for (int i = 0; i < n; ++i)
            ret[athletes[i].pos] = j <= 2 ? medal[j++] : to_string(i + 1);
        return ret;

    }
};

File: LeetCode/python3/506.py
class Solution:
    class Athlete:
        def __init__(self, score, pos):
            self.score = score
            self.pos = pos

    def findRelativeRanks(self, nums: List[int]) -> List[str]:
        n, athletes = len(nums), []
        for i in range(n):
            athletes.append(self.Athlete(nums[i], i))
        athletes = sorted(athletes, key=lambda a: a.score, reverse=True)
        ret = ["" for _ in range(n)]
        medal = ["Gold Medal", "Silver Medal", "Bronze Medal"]
        i, j = 0, 0
        for i in range(n):
            if j <= 2:
                ret[athletes[i].pos] = medal[j]
                j += 1
            else:
                ret[athletes[i].pos] = str(i + 1)
            i += 1
        return ret
